src/nvim/ops.c

```diff
@@ -3644,6 +3644,10 @@ int do_join(size_t count,
    */
   for (t = (linenr_T)count - 1;; t--) {
     cend -= currsize;
+	if(cend == NULL || curr == NULL){
```

Comment:
As I was trying this out, `exec make -C build unittest` closes my terminal. Ever encounter this issue before? I retested on a vanilla copy of the code and ran into the same issue.